pkg/controllers/metrics/pod/controller.go

```diff
@@ -88,13 +105,15 @@ func labelNames() []string {
 // NewController constructs a controller instance
 func NewController(kubeClient client.Client) *Controller {
 	return &Controller{
-		kubeClient: kubeClient,
+		kubeClient:  kubeClient,
+		pendingPods: sets.NewString(),
 	}
 }
 
 // Reconcile executes a termination control loop for the resource
 func (c *Controller) Reconcile(ctx context.Context, req reconcile.Request) (reconcile.Result, error) {
 	ctx = logging.WithLogger(ctx, logging.FromContext(ctx).Named("podmetrics").With("pod", req.Name))
+	ts := time.Now()
```

Comment:
Watch events are eventually consistent. Can we use the timestamp on a pod's status condition?

Reply:
Yep, I didn't think there was a recorded timestamp but there is a `LastTransitionTime` field that looks like it should do the trick.